src/sonic-bgpcfgd/app/allow_list.py

```diff
@@ -0,0 +1,535 @@
+"""
+Implementation of "allow-list" feature
+"""
+import re
+
+from app.log import log_debug, log_info, log_err, log_warn
+from app.template import TemplateFabric
+from app.manager import Manager
+from app.util import run_command
+
+class BGPAllowListMgr(Manager):
+    """ This class initialize "AllowList" settings """
+    ALLOW_ADDRESS_PL_NAME_TMPL = "ALLOW_ADDRESS_%d_%s"  # template for a name for the ALLOW_ADDRESS prefix-list ???
+    EMPTY_COMMUNITY = "empty"
+    PL_NAME_TMPL = "PL_ALLOW_LIST_DEPLOYMENT_ID_%d_COMMUNITY_%s_V%s"
+    COMMUNITY_NAME_TMPL = "COMMUNITY_ALLOW_LIST_DEPLOYMENT_ID_%d_COMMUNITY_%s"
```

Comment:
how do you calculate the routemap order for this route map? where is the code to generate the order?

Reply:
I maintain only one order here:
1. All route-maps without community strings are placed after 29999 route entry, and all route-maps with community strings are places in range 1-29999. This is to ensure that we will not match and routes without community value, when we have a rules with a community value.